pkg/synthetictests/historicaldata/types.go

```diff
@@ -120,14 +129,21 @@ func (b *bestMatcher) BestMatch(name string, jobType platformidentification.JobT
 		nil
 }
 
-func (b *bestMatcher) BestMatchDuration(name string, jobType platformidentification.JobType) (StatisticalDuration, string, error) {
-	rawData, details, err := b.BestMatch(name, jobType)
+func (b *bestMatcher) BestMatchDuration(name string, jobType platformidentification.JobType, extraMatchers ...NextBestKey) (StatisticalDuration, string, error) {
+	rawData, details, err := b.BestMatch(name, jobType, extraMatchers...)
 	return toStatisticalDuration(rawData), details, err
 }
 
-func (b *bestMatcher) BestMatchP99(name string, jobType platformidentification.JobType) (*time.Duration, string, error) {
-	rawData, details, err := b.BestMatchDuration(name, jobType)
-	return &rawData.P99, details, err
+func (b *bestMatcher) BestMatchPercentile(p Percentile, name string, jobType platformidentification.JobType, extraMatchers ...NextBestKey) (*time.Duration, string, error) {
+	rawData, details, err := b.BestMatchDuration(name, jobType, extraMatchers...)
+	switch p {
+	case P95:
+		return &rawData.P95, details, err
+	case P99:
+		return &rawData.P99, details, err
+	default:
+		return &rawData.P99, details, fmt.Errorf("unsupported percentile of (%s)", p)
```

Comment:
Nit: probably would expect nil returned here for the percentile when there's an error.